src/Nethermind/Nethermind.Evm/EvmStack.cs

```diff
@@ -13,165 +13,178 @@
             }
             else
             {
-                value.Span.CopyTo(word);
+                Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(word), Unsafe.As<byte, Word>(ref MemoryMarshal.GetReference(value.Span)));
             }
 
             if (++Head >= MaxStackSize)
             {
-                Metrics.EvmExceptions++;
-                ThrowEvmStackOverflowException();
+                EvmStack.ThrowEvmStackOverflowException();
             }
         }
 
+        public ref byte PushBytesRef()
+        {
+            ref byte bytes = ref _bytes[Head * WordSize];
+
+            if (++Head >= MaxStackSize)
+            {
+                EvmStack.ThrowEvmStackOverflowException();
+            }
+
+            return ref bytes;
+        }
+
         public void PushBytes(scoped in ZeroPaddedMemory value)
         {
-            if (_tracer.IsTracingInstructions) _tracer.ReportStackPush(value);
+            if (typeof(TTracing) == typeof(IsTracing)) _tracer.ReportStackPush(value);
 
-            Span<byte> word = _bytes.Slice(Head * 32, 32);
-            if (value.Memory.Length != 32)
+            Span<byte> word = _bytes.Slice(Head * WordSize, WordSize);
+            if (value.Memory.Length != WordSize)
             {
-                word.Clear();
+                // Clear the word first
+                Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(word), Word.Zero);
                 value.Memory.Span.CopyTo(word[..value.Memory.Length]);
             }
             else
             {
-                value.Memory.Span.CopyTo(word);
+                Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(word), Unsafe.As<byte, Word>(ref MemoryMarshal.GetReference(value.Memory.Span)));
             }
 
             if (++Head >= MaxStackSize)
             {
-                Metrics.EvmExceptions++;
-                ThrowEvmStackOverflowException();
+                EvmStack.ThrowEvmStackOverflowException();
             }
         }
 
         public void PushByte(byte value)
         {
-            if (_tracer.IsTracingInstructions) _tracer.ReportStackPush(value);
+            if (typeof(TTracing) == typeof(IsTracing)) _tracer.ReportStackPush(value);
 
-            Span<byte> word = _bytes.Slice(Head * 32, 32);
-            word.Clear();
-            word[31] = value;
+            Span<byte> word = _bytes.Slice(Head * WordSize, WordSize);
+            // Clear the word first
+            Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(word), Word.Zero);
+            word[WordSize - sizeof(byte)] = value;
 
             if (++Head >= MaxStackSize)
             {
-                Metrics.EvmExceptions++;
-                ThrowEvmStackOverflowException();
+                EvmStack.ThrowEvmStackOverflowException();
             }
         }
 
-        private static readonly byte[] OneStackItem = { 1 };
+        private static ReadOnlySpan<byte> OneStackItem() => new byte[] { 1 };
```

Comment:
won't this create a new array everytime it gets invoked ?

Reply:
No is an optimization for `ReadOnlySpan<byte>` statics; as the data cannot ever be modified the array gets baked into the data section of the dll and the span points directly at that rather than at an in memory instantiation